spec/models/green_flag/feature_spec.rb

```diff
@@ -1,135 +1,187 @@
-        it 'returns a feature decision with the correct attributes' do
-          feature.stub(rules: [non_applying_rule, true_rule])
+      it 'should not create a feature event' do
+        expect(GreenFlag::FeatureEvent.count).to eq 0
+      end
+    end
 
-          feature_decision = feature.decide_if_enabled_for_visitor(site_visitor_id)
+    context 'when there are manual decisions' do
+      subject { feature.forget_non_manual_decisions!(enabled) }
 
-          expect(feature_decision.rule_id).to eq true_rule.id
-          expect(feature_decision.enabled).to eq true
-        end
+      before(:each) do
+        @manual_fd = GreenFlag::FeatureDecision.create!(feature_id: feature.id, manual: true, site_visitor_id: 1)
+      end
+
+      it 'should not delete the manual decisions' do
+        expect(GreenFlag::FeatureDecision.find_by_id(@manual_fd.id)).to be_present
       end
 
-      context 'the first applying rule is false' do
-        it 'returns a feature decision with the correct attributes' do
-          feature.stub(rules: [false_rule, true_rule])
+      it 'should not create a feature event' do
+        expect(GreenFlag::FeatureEvent.count).to eq 0
+      end
+    end
 
-          feature_decision = feature.decide_if_enabled_for_visitor(site_visitor_id)
+    context 'when there are automatic decisions' do
+      before(:each) do
+        @feature2 = GreenFlag::Feature.create!(code: 'huge_banner_ad')
+        @auto_fd = GreenFlag::FeatureDecision.create!(feature_id: @feature2.id, manual: false, enabled: true, site_visitor_id: 1)
 
-          expect(feature_decision.rule_id).to eq false_rule.id
-          expect(feature_decision.enabled).to eq false
-        end
+        @feature2.forget_non_manual_decisions!(enabled)
+      end
+
+      it 'should delete the automatic decisions' do
+        @feature2.forget_non_manual_decisions!(enabled)
+        expect(GreenFlag::FeatureDecision.find_by_id(@auto_fd)).to be_nil
+      end
+
+      it 'should create a feature event with the correct attributes' do
+        feature_event = GreenFlag::FeatureEvent.last
+        expect(feature_event).to be_present
+        expect(feature_event.feature_id).to eq @feature2.id
+        expect(feature_event.event_type_code).to eq GreenFlag::FeatureEvent::ENABLED_DECISIONS_FORGOTTEN
+        expect(feature_event.count).to eq 1
       end
     end
+  end
 
-    describe '#forget_non_manual_decisions!' do
-      let(:feature) { GreenFlag::Feature.create(code: 'big_font') }
-      let(:enabled) { true }
+  describe "#latest_version" do
+    let(:feature) { FactoryGirl.create(:green_flag_feature, version_number: 1) }
 
-      context 'when there are no decisions' do
-        subject { feature.forget_non_manual_decisions!(enabled) }
+    subject { feature.latest_version }
 
-        it 'should not attempt to delete non-manual decisions' do
-          expect(GreenFlag::FeatureDecision.non_manual).to receive(:delete_all).never
+    context 'when there are rules' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.